### PR TITLE
add gsets support

### PR DESCRIPTION
**Pull request by binarytemple-bet365**
@russelldb another gsets pull request

### CODE REVIEW

File: src/riak_dt.proto [hunk 1]
```diff
@@ -107,6 +107,7 @@ message DtValue {
     optional sint64   counter_value = 1;
     repeated bytes    set_value     = 2;
     repeated MapEntry map_value     = 3;
+    repeated bytes    gset_value    = 4;
```
Comment (lukebakken) [inline src/riak_dt.proto]: Re-using `set_value` should be possible. I don't see the benefit of a separate field as `DataType` disambiguates it.

File: src/riak_dt.proto [hunk 2]
```diff
@@ -154,6 +156,18 @@ message SetOp {
     repeated bytes removes = 2;
 }
 
+
+/*
+ * An operation to update a GSet, on its own.
+ * GSet members are opaque binary values, you can only add
+ * them to a Set.
+ */
+message GSetOp {
```
Comment (lukebakken) [inline src/riak_dt.proto]: `SetOp` can be re-used. If `removes` are part of a `SetOp` for `GSET`, Riak should return an error.

File: src/riak_dt.proto [hunk 3]
```diff
@@ -205,6 +220,7 @@ message DtOp {
     optional CounterOp counter_op = 1;
     optional SetOp     set_op     = 2;
     optional MapOp     map_op     = 3;
+    optional GSetOp    gset_op    = 4;
```
Comment (lukebakken) [inline src/riak_dt.proto]: `set_op` can be re-used

File: src/riak_dt.proto [hunk 4]
```diff
@@ -251,4 +267,6 @@ message DtUpdateResp {
     optional sint64   counter_value = 3;
     repeated bytes    set_value     = 4;
     repeated MapEntry map_value     = 5;
-}
\ No newline at end of file
+    repeated bytes    gset_value    = 6;
```
Comment (lukebakken) [inline src/riak_dt.proto]: `set_value` can be re-used

**Comment by binarytemple-bet365** [comment]
@lukebakken I need to close this P/R and re-target. Will address your comments in https://github.com/basho/riak_pb/pull/203

**Comment by bashopatricia** [comment]
create jira issue

**Pull request closed without merging**